src/jit/morph.cpp

```diff
@@ -10309,13 +10309,11 @@ GenTree* Compiler::fgMorphCopyBlock(GenTree* tree)
 
         JITDUMP(requiresCopyBlock ? " this requires a CopyBlock.\n" : " using field by field assignments.\n");
 
-        // Mark the dest/src structs as DoNotEnreg
-        // when they are not reg-sized non-field-addressed structs and we are using a CopyBlock
-        // or the struct is not promoted
+        // Mark the dest/src structs as DoNotEnreg when they are not being fully referenced.
```

Comment:
Should this clarify that this also includes when the src/dest are not the same type (in some of these cases, the destination struct and register are the same size, and I would guess that implies the destination struct is fully referenced).